### PR TITLE
Implemented changes in Zvtx and SPD-vertex-based cuts in IsEventSelected

### PR DESCRIPTION
**Pull request by fgrosa**
1) swapped order of Zvtx and SPD-vertex-based cuts
2) set default of flag fCutOnzVertexSPD=3
3) added flag to apply |Zvtx|>X cm cut on SPD vertex

### CODE REVIEW

File: PWGHF/vertexingHF/AliRDHFCuts.cxx [hunk 1]
```diff
@@ -632,52 +634,65 @@ Bool_t AliRDHFCuts::IsEventSelected(AliVEvent *event) {
...
+          }
+        }
       }
     }
   }
 
+  Double_t zvert = -999;
+  if(!fApplyZcutOnSPDvtx && vertex) zvert=vertex->GetZ();
+  else if(fApplyZcutOnSPDvtx) {
+    if(!vSPD){
```
Comment (fprino) [inline PWGHF/vertexingHF/AliRDHFCuts.cxx]: Hi Fabrizio,

thanks for this implemtentation.
Just one doubt: at this point shouldn't we also check that the contributors to the SPD vertex are >0?
Cheers, Francesco

File: PWGHF/vertexingHF/AliRDHFCuts.cxx [hunk 2]
```diff
@@ -632,52 +634,65 @@ Bool_t AliRDHFCuts::IsEventSelected(AliVEvent *event) {
...
+      accept=kFALSE;
+      fEvRejectionBits+=1<<kBadSPDVertex;
+    }
+    else zvert = vSPD->GetZ();
+  }
+  
+  if((!fApplyZcutOnSPDvtx && vertex) || (fApplyZcutOnSPDvtx && vSPD)) {
```
Comment (fprino) [inline PWGHF/vertexingHF/AliRDHFCuts.cxx]: Do we need this if condition?
if the conditions (!fApplyZcutOnSPDvtx && vertex) || (fApplyZcutOnSPDvtx && vSPD) are not satisfied, then zvert is -999 and will not enter in the fMaxVtxZ region.

File: PWGHF/vertexingHF/AliRDHFCuts.cxx (hunk 1 above)
Comment (fgrosa) [reply PWGHF/vertexingHF/AliRDHFCuts.cxx]: Hi Francesco, 
right! I forgot it, I add it now with an additional commit. Thanks!

File: PWGHF/vertexingHF/AliRDHFCuts.cxx (hunk 2 above)
Comment (fgrosa) [reply PWGHF/vertexingHF/AliRDHFCuts.cxx]: Yes, you are right! I remove it in the new commit. Thanks again!

**Comment by fprino** [comment]
+1
Thanks for implementing my suggesstions!

**Pull request merged**